error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/MockitoMockClassReferenceTest.java

```diff
@@ -0,0 +1,139 @@
+package tech.picnic.errorprone.bugpatterns;
+
+import static com.google.errorprone.BugCheckerRefactoringTestHelper.TestMode.TEXT_MATCH;
+
+import com.google.errorprone.BugCheckerRefactoringTestHelper;
+import com.google.errorprone.CompilationTestHelper;
+import org.junit.jupiter.api.Test;
+
+final class MockitoMockClassReferenceTest {
+  private final CompilationTestHelper compilationTestHelper =
+      CompilationTestHelper.newInstance(MockitoMockClassReference.class, getClass());
+  private final BugCheckerRefactoringTestHelper refactoringTestHelper =
+      BugCheckerRefactoringTestHelper.newInstance(MockitoMockClassReference.class, getClass());
+
+  @Test
+  void identification() {
+    compilationTestHelper
+        .addSourceLines(
+            "A.java",
+            "import static org.mockito.Mockito.mock;",
+            "import static org.mockito.Mockito.spy;",
+            "",
+            "import java.util.List;",
+            "",
+            "class A {",
+            "  // BUG: Diagnostic contains:",
+            "  String memberMock = mock(String.class);",
+            "  // BUG: Diagnostic contains:",
+            "  String memberSpy = mock(String.class);",
+            "",
+            "  void m() {",
+            "    Integer variableMock;",
+            "    // BUG: Diagnostic contains:",
+            "    variableMock = mock(Integer.class);",
+            "    // BUG: Diagnostic contains:",
+            "    List nonGenericallyTypedMock = mock(List.class);",
+            "    var dynamicallyTypedMock = mock(Integer.class);",
+            "    Integer namedMock = mock(Integer.class, \"name\");",
+            "    Object subtypeMock = mock(Integer.class);",
+            "",
+            "    // BUG: Diagnostic contains:",
+            "    String equalTypedSpy = spy(String.class);",
+            "    // BUG: Diagnostic contains:",
+            "    List nonGenericallyTypedSpy = spy(List.class);",
+            "    var dynamicallyTypedSpy = spy(Integer.class);",
+            "    Object objectSpy = spy(new Object());",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  void unimplementedCases() {
+    // XXX: Move to identification once fixed.
+    compilationTestHelper
+        .addSourceLines(
+            "A.java",
+            "import static org.mockito.Mockito.mock;",
+            "",
+            "import java.math.BigInteger;",
+            "import java.util.List;",
+            "",
+            "class A {",
+            "  void m() {",
+            "    // This case is arguable as it is unsafe in any case.",
+            "    // This currently is not identified as the `erasure` is the same.",
+            "    List<String> genericallyTypedMock = mock(List.class);",
```

Comment:
As pointed out, this is an unsafe cast in both before and after case. As such, at least in our own codebases, we sometimes have casts with `SuppressWarnings`. If we are to rewrite these and we can circumvent these warnings, it might be nice to also remove existing suppressions.

Reply:
Hmm, I would have expected dropping `List.class` to be safe here. And if we can drop a suppression in this case: great!

Reply:
I don't think we should rewrite this case, if we would, it'd require a suppression as you mention. That is not an improvement IMO, so I wouldn't bother for now.

Reply:
The same holds for the `getGenericMock` so I'm not sure if we would gain anything from that. Also, if we apply that change, then the parameter would be unused 🤔. So all in all, not sure if it would be an improvement.

Reply:
I filed mockito/mockito#2866; with those changes dropping `List.class` would be fine.